pkgs/development/tools/pandoc/default.nix

```diff
@@ -13,6 +13,9 @@ in
       remove-references-to \
         -t ${haskellPackages.pandoc-types} \
         $out/bin/pandoc
+      remove-references-to \
+        -t ${haskellPackages.warp} \
+        $out/bin/pandoc
```

Comment:
Why does pandoc depend on references to warp to begin with? Please add the reason this is safe to do as a comment.

Reply:
By using `justStaticExecutables foo`,  the compiled executable shouldn't refer to any Haskell libraries and documentation. However, this is not always the case when the dependency has `Paths_` module - `bindir`, `datadir`, and `libdir` which contain the path to that Haskell library will still appear in the executable (https://github.com/NixOS/nixpkgs/issues/34376#issuecomment-375745951). Most Haskell programs only use `Paths_foo.version` for getting its version at runtime, while not using those directories, so it's safe to remove the reference to the Haskell library. And this is exactly the case of `warp`:

```haskell
Network/Wai/Handler/Warp/Settings.hs
21:import qualified Paths_warp
195:    , settingsServerName = C8.pack $ "Warp/" ++ showVersion Paths_warp.version

Network/Wai/Handler/Warp/Response.hs
34:import qualified Paths_warp
409:warpVersion = showVersion Paths_warp.version
```

Reply:
And there is already a comment at line 25 that explains everything.

Reply:
Could you copy paste your reply 1:1 into a comment? Context like this helps a lot when someone looks at this again in the future.